frontend/src/test/scala/bloop/ScalaVersionsSpec.scala

```diff
@@ -9,87 +9,85 @@ import bloop.util.TestProject
-          } else {
-            """/main/scala/Foo.scala
-              |class Foo
+        } else {
+          """/main/scala/Foo.scala
+            |class Foo
             """.stripMargin
-          }
         }
+      }
 
-        val logger = new RecordingLogger(ansiCodesSupported = false)
-        loggers.synchronized { loggers = logger :: loggers }
-        val jars = jarsForScalaVersion(scalaVersion, logger)
-        val `A` = TestProject(
-          workspace,
-          "a",
-          List(source),
-          scalaOrg = Some(compilerOrg),
-          scalaCompiler = Some(compilerArtifact),
-          scalaVersion = Some(scalaVersion),
-          jars = jars
-        )
+      val jars = jarsForScalaVersion(scalaVersion, logger)
+      val `A` = TestProject(
+        workspace,
+        "a",
+        List(source),
+        scalaOrg = Some(compilerOrg),
+        scalaCompiler = Some(compilerArtifact),
+        scalaVersion = Some(scalaVersion),
+        jars = jars
+      )
 
-        val projects = List(`A`)
-        val state = loadState(workspace, projects, logger)
-        val compiledState = state.compile(`A`)
-        try {
-          Predef.assert(compiledState.status == ExitStatus.Ok)
-          assertValidCompilationState(compiledState, projects)
-        } finally loggers.synchronized { loggers = loggers.filterNot(_ == logger) }
-      }
+      val projects = List(`A`)
+      val state = loadState(workspace, projects, logger)
+      val compiledState = state.compile(`A`)
+      Predef.assert(compiledState.status == ExitStatus.Ok)
+      assertValidCompilationState(compiledState, projects)
     }
+  }
 
-    val `2.10` = compileProjectFor("2.10.7")
-    val `2.11` = compileProjectFor("2.11.12")
-    val `2.12` = compileProjectFor("2.12.9")
-    val `2.13` = compileProjectFor("2.13.2")
-    val `2.13.3` = compileProjectFor("2.13.3")
-    val `2.13.7` = compileProjectFor("2.13.7")
-    val `2.13.8` = compileProjectFor("2.13.8")
-    val LatestDotty = compileProjectFor("3.0.0-M3")
-    val all = {
-      if (TestUtil.isJdk8)
-        List(`2.10`, `2.11`, `2.12`, `2.13`, `2.13.3`, `2.13.7`, `2.13.8`, LatestDotty)
-      else List(`2.12`, `2.13`, `2.13.3`, `2.13.7`)
-    }
+  val jdk8OnlyVersions = List(
+    "2.10.7",
+    "2.11.12"
+  )
+  val scalaVersions = List(
+    "2.12.16",
+    "2.13.6",
+    "2.13.7",
+    "2.13.8",
+    "3.0.2",
+    "3.1.3"
+  )
 
-    try {
-      TestUtil.await(FiniteDuration(120, "s"), ExecutionContext.ioScheduler) {
-        Task
-          .sequence(all.grouped(2).map(group => Task.gatherUnordered(group)))
-          .map(_ => ())
-      }
-    } catch {
-      case NonFatal(t) =>
-        loggers.foreach(logger => logger.dump())
-        Thread.sleep(100)
-        System.err.println(TestUtil.threadDump)
-        Thread.sleep(100)
-        throw t
-    } finally {}
+  val allVersions = if (TestUtil.isJdk8) jdk8OnlyVersions ++ scalaVersions else scalaVersions
+
+  allVersions.foreach { scalaV =>
```

Comment:
Perhaps controversially, I've changed how this works.

Previously, it would run tests for all Scala versions as part of a single test, grouping by two and running them in parallel. If something failed, you'd get no indication as to which Scala version was used. 
The upside that the tests were running in parallel. 

I've changed them to run sequentially and report as individual tests. This does take longer. I will be happy to change it back and just port the small changes I've made, namely:

1. Modern Scala 2 and 3 versions
2. Correctly resolve Scala 3 artifacts
3. don't use `synchronized` and give each test its own logger.

Let me know.

Reply:
It's fine to make it work sequential, some flakiness might have come from that :thinking: